### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by nschloe**
A couple of high-impact fixes from Coverity, mostly resource leaks and uninitialized variables.

### CODE REVIEW

**Comment by rryan** [comment]
Thanks @nschloe  -- code in lib and vamp-plugins isn't written by us so we don't do mass reformatting to aid with merging new revisions. Could you revert the trailing whitespace changes? Looks good otherwise.

**Comment by daschuer** [comment]
We may consider to keep the fidlib changes and switch entirely to https://github.com/uazu/fidlib/pull/1
It is just on the way to be merged to upstream (since 5 month .. )
This way we get rid of the annoying compiler warnings as well.  
What do you think?

File: vamp-plugins/dsp/TempoTrack.cpp [hunk 1]
```diff
@@ -544,7 +544,7 @@ int TempoTrack::findMeter(double *ACF, unsigned int len, double period)
 	tsig = 4;
     }
 
-
+    delete(dbf);
```
Comment (daschuer) [inline vamp-plugins/dsp/TempoTrack.cpp]: It looks like dbf is unused. Can we delete it? 
We should provide a patch upstream as well.

File: src/library/rhythmbox/rhythmboxfeature.cpp [hunk 2]
```diff
@@ -258,6 +258,7 @@ TreeItem* RhythmboxFeature::importPlaylists() {
         // do error handling
         qDebug() << "Cannot process Rhythmbox music collection";
         qDebug() << "XML ERROR: " << xml.errorString();
+        delete(rootItem);
```
Comment (daschuer) [inline src/library/rhythmbox/rhythmboxfeature.cpp]: Nitpick (sorry): we prefer 
delete rootItem;

**Comment by nschloe** [comment]
@rryan Done & done.

**Comment by nschloe** [comment]
@daschuer Sounds good. In future, to avoid such long waiting times for these fixes, I think we could even just use your (or Mixxx's) fork of a third-party library.

**Comment by nschloe** [comment]
(Btw, if this gets merged before Friday night (CET), we should see an improvement in the defects stats for the weekly Coverity check.)

File: src/widget/wtracktableviewheader.cpp [hunk 3]
```diff
@@ -42,7 +42,11 @@ HeaderViewState::HeaderViewState(const QString& base64serialized) {
     array.append(base64serialized);
     // First decode the array from Base64, then initialize the protobuf from it.
     array = QByteArray::fromBase64(array);
-    m_view_state.ParseFromArray(array.constData(), array.size());
+    if (!m_view_state.ParseFromArray(array.constData(), array.size())) {
+      qDebug() << "ERROR: Could not parse m_view_state from QByteArray of size "
```
Comment (daschuer) [inline src/widget/wtracktableviewheader.cpp]: Please check for 4 space indent in this PR and 8 space or other non 4 Space indent for line breaks 
"qDebug()::operator<<" connects the strings and adds a space for us.

**Comment by daschuer** [comment]
LGTM. Thank you very much!

**Pull request merged**